### ISSUE TITLE
Final Access Modifier Kotlin

### ISSUE THREAD

**Issue by timrijckaert** [Col-E/Recaf#99]
Hi, 

First of very nice work. 
Nice UI very simple to go through. The alternatives are not even on par with this lib.
This project could use some more recognition imho.

My question/bug:
I don't know if Recaf was only targeted for applications written in pure Java or not.
However I'm facing an issue when trying to edit my access modifier of a Kotlin written .class

Consider the following Kotlin class  

```
class Main {
	fun main(args: Array<String>) {
		println("Hello world from Kotlin")
	}
}
```
Since every class in Kotlin is `final` by default this `Main` class is too.  
Recaf successfully reports this aswell.

![screenshot from 2018-10-14 22-21-28](https://user-images.githubusercontent.com/6739443/46921793-f93dff00-cfff-11e8-8170-ecc73c892419.png)

However upon removing the `final` modifier, saving it and trying to extend it

```
class SomeExtendedClass : Main() {
	fun someOtherMethod() {
			println("Some Other Message")
	}
}
```
Throws upon compilation:
> SomeExtendedClass.kt:1:28: error: this type is final, so it cannot be inherited from
> class SomeExtendedClass : Main()

However the equivalent Java code does seem to do the trick.

```
final class Main {
	public static void main(String[] args) {
		System.out.println("Hello this is a Java world!");
	}
}
```

So first of, is it possible to edit the byte code from a Kotlin class?
And if so how can I remove a `final` modifier and make it `open/public`

**Comment by Col-E** [comment]
Thanks for the kind words!

As for the issue, it should_work the same regardless of the source language. I'll take a look at it when I'm not swamped with exams / class work.

**Comment by Col-E** [comment]
> Kotlin uses an annotation called Metadata to store some extra information within the compiled class, and it's used by the **Kotlin compiler** to determine a compiled Kotlin class's actual modifier. Therefore, only change the modifiers specified in JVMS is not enough, you have to modify the data held in Metadata annotation too.

Annotation editing support isn't fully fleshed out yet, so that's a TODO right there. For now, would it be possible to delete this annotation? I assume this would make the kotlin compiler treat the modified class as a *standard javac-compiled* class rather than one from the *kotlin compiler*. If that is the case then it should go back to using the access modifiers.

**Comment by timrijckaert** [comment]
How would I remove the annotation with Recaf?

**Comment by Col-E** [comment]
So if I open a method that looks like the following:
```java
    @Listener(priority=-1)
    private void onNewInput(NewInputEvent input) {
          // code
    }
```
You can select the visible annotations and it will show a list. Select the item and hit the `DELETE` key.

![image](https://user-images.githubusercontent.com/21371686/47541011-9ec95c00-d8a5-11e8-8497-da4178c39bd5.png)

**Comment by timrijckaert** [comment]
No problem.
I have tried your solution.

The `@Metadata` annotation is in the **Visible annotations** menu.
However it is complete gibberish.

![screenshot from 2018-10-26 10-59-51](https://user-images.githubusercontent.com/6739443/47556315-49756500-d90e-11e8-88de-d471729f41be.png)

I tried deleting it and exporting it.
Whilst also setting the access to public

The JD-GUI decompiler now successfully says this class is a `public class` however in Android Studio the class is still final.
I think deleting is not the solution. I probably need to edit the `@Metadata` annotation to tell it this class should be `open` 

![screenshot from 2018-10-26 11-00-36](https://user-images.githubusercontent.com/6739443/47556343-627e1600-d90e-11e8-8748-02c0484c0c4b.png)
![screenshot from 2018-10-26 11-04-17](https://user-images.githubusercontent.com/6739443/47556548-e6380280-d90e-11e8-8605-4290de372b35.png)
